### ISSUE TITLE
3.1.10 and 3.1.11 fails to catch no-unused-vars

### ISSUE THREAD

**Issue by PiPeep** [babel/babel-eslint#120]
- Minimal testcase: https://github.com/PiPeep/babel-eslint-testcase
- Impacts: facebook/react#4032

@zpao @spicyj

**Comment by Dashed** [comment]
Having the same issue. 

`no-unused-vars` actually partially works for this sample:

```js
var x = 10;

var y = 10;
y = 5;

// By default, unused arguments cause warnings.
(function(foo) {
    return 5;
})();

// Unused recursive functions also cause warnings.
function fact(n) {
    if (n < 2) return 1;
    return n * fact(n - 1);
}
```

Output:

```
   7:10  error  foo is defined but never used    no-unused-vars
  12:9   error  fact is defined but never used   no-unused-vars
```

**Comment by hzoo** [comment]
Ah that sucks.. I guess trying to fix the `no-unused-vars` flow issues (#108) in 3.1.10 led to these issues. Will fix. Adding the unit test for this will prevent me from messing that up again!

**Comment by hzoo** [comment]
Ok hopefully fixed in 3.1.13. @PiPeep @zpao @spicyj. (Erroring correctly now in your example)

**Comment by Dashed** [comment]
The fix works. But now catches used variables for JSX.

```js
const React = require("react");

const Component = React.createClass({
    render() {
        return (<div></div>);
    }
});

React.render(<Component />);
```

Output:

```js
✗ 1 albertoleal babel-eslint-testcase $ ./node_modules/.bin/eslint testcase.js

testcase.js
  3:6  error  Component is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

Testcase: https://github.com/Dashed/babel-eslint-testcase

**Comment by sebmck** [comment]
@Dashed Are you using [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react)? This has always been the case/behaviour, see #110, #79, #6 and #5.

**Comment by hzoo** [comment]
Yeah what @sebmck said haha.

**Comment by Dashed** [comment]
Heh. I missed that part on the README. Using `eslint-plugin-react` now works. :+1:
